From 679fa83791c25c43014da65eea842a374ccd28de Mon Sep 17 00:00:00 2001
From: Isaac Boukris <iboukris@gmail.com>
Date: Thu, 30 Dec 2021 17:26:17 +0200
Subject: [PATCH 1/2] TEMP: patch SUBDIRS to only include a subset

---
 src/Makefile.in | 27 +--------------------------
 1 file changed, 1 insertion(+), 26 deletions(-)

diff --git a/src/Makefile.in b/src/Makefile.in
index 8f14e9bf2..31272f45e 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -6,32 +6,7 @@ mydir=.
 #	plugins/locate/python
 #	plugins/preauth/wpse
 #	plugins/preauth/cksum_body
-SUBDIRS=util include lib \
-	@sam2_plugin@ \
-	plugins/audit \
-	plugins/audit/test \
-	@audit_plugin@ \
-	plugins/kadm5_hook/test \
-	plugins/kadm5_auth/test \
-	plugins/gssapi/negoextest \
-	plugins/hostrealm/test \
-	plugins/localauth/test \
-	plugins/pwqual/test \
-	plugins/authdata/greet_server \
-	plugins/authdata/greet_client \
-	plugins/certauth/test \
-	plugins/kdb/db2 \
-	@ldap_plugin_dir@ \
-	@lmdb_plugin_dir@ \
-	plugins/kdb/test \
-	plugins/kdcpolicy/test \
-	plugins/preauth/otp \
-	plugins/preauth/pkinit \
-	plugins/preauth/spake \
-	plugins/preauth/test \
-	plugins/tls/k5tls \
-	kdc kadmin kprop clients appl tests \
-	config-files build-tools man doc @po@
+SUBDIRS=util include lib
 WINSUBDIRS=include util lib ccapi windows clients appl plugins\preauth\spake
 BUILDTOP=$(REL).
 
-- 
2.31.1
